### PR TITLE
RFC 2606 and  RFC 5737 fully applied. Other changes from #92

### PR DESCRIPTION
**Pull request by pawel-kow**
Completed the changes discussed in #92.

Fixed other issues like:
asciidoctor: ERROR: Domain Connect Spec Draft.adoc: line 733: dropping cells from incomplete row detected end of table
asciidoctor: ERROR: Domain Connect Spec Draft.adoc: line 1090: dropping cells from incomplete row detected end of table

### CODE REVIEW

**Comment by kerolasa** [comment]
It would be nice to get `4.2.3. Security Considerations ` improvement to signature generation. Something like this:

```
Notes:

The digital signature will be generated on the query string.  The values of
each query string value key/value pair MUST be properly URL Encoded before
the signature is generated.  The sig and key parameters MUST be excluded
from the query string before generating signature.
```

p.s. That was incorrectly implemented in Cloudflare. When I get the next release in use it will have a counter how often
ServiceProviders are sending signatures that include sig or key. It will be interesting to see how many broken
implementations there are due to missing a note.

File: Domain Connect Spec Draft.adoc [hunk 1]
```diff
@@ -622,31 +704,31 @@ be used when applying the template.
...
-|(optional) This parameter allows for the caller to provide additional text for display
+|(OPTIONAL) This parameter allows for the caller to provide additional text for display
 with the template serviceName. It should be used to augment the serviceName value
 from the template, not replace it. This parameter is only allowed when the
 "sharedServiceName" attribute is set in the template.
 
 |*Group Id*
 |groupId
-|(optional) This parameter specifies the groups from the template to apply. 
+|(OPTIONAL) This parameter specifies the groups from the template to apply.
```
Comment (kerolasa) [inline Domain Connect Spec Draft.adoc:722]: Nit. There are some end of line white spaces that can be removed.

    sed -i 's/ *^M$/^M/' Domain\ Connect\ Spec\ Draft.adoc

File: Domain Connect Spec Draft.adoc [hunk 2]
```diff
@@ -1067,7 +1147,7 @@ token.
 
 |*Redirect URI*
 |redirect_uri
-|(optional) This is required if a redirect_uri was
+|(OPTIONAL) This is required if a redirect_uri was
```
Comment (kerolasa) [inline Domain Connect Spec Draft.adoc:1152]: Perhaps all the tables should have RFC2119 colum that tells: REQUIRED, RECOMMENDED, or OPTIONAL.

File: Domain Connect Spec Draft.adoc [hunk 3]
```diff
@@ -65,6 +65,8 @@ DNS Providers and Service Providers to simplify this interaction across the inte
 [glossary]
 === Terminology
 
+The key words "*MUST*", "*MUST NOT*", "*REQUIRED*", "*SHALL*", "*SHALL NOT*", "*SHOULD*", "*SHOULD NOT*", "*RECOMMENDED*", "*NOT RECOMMENDED*", "*MAY*", and "*OPTIONAL*" in this document are to be interpreted as described in BCP 14 <<RFC2119>> <<RFC8174>> when, and only when, they appear in all capitals, as shown here.
+
 [horizontal]
 Service Providers:: refers to entities that provide products and
 services attached to domain names. Examples include web hosting
```
Comment (kerolasa) [inline Domain Connect Spec Draft.adoc:72]: Couple lines after this: s/CloudFlare/Cloudflare/

**Review by kerolasa — APPROVED** [review]
Lots of good changes in this pull. Feel free to add a commit to address my feedback and merge these.

File: Domain Connect Spec Draft.adoc [hunk 4]
```diff
@@ -1407,25 +1485,20 @@ aware that their access to the API may be denied.
...
-Any changes to the template need to account for existing customers with settings in 
-DNS, some applied through Domain Connect and some manual. So when changes are made, 
-they are often backward compatible. 
+Any changes to the template need to account for existing customers with settings in DNS, some applied through Domain Connect and some manual. So when changes are made, they are often backward compatible. 
 
 Note that when a template changes, it does need to be on-boarded with the DNS Providers.
 
-The <<template-version-field, version field>> of the template definition serves the purpose
-of transparency between the DNS Provider and the Service Provider in case of such changes.
+The <<template-definition, version field>> of the template definition serves the purpose of transparency between the DNS Provider and the Service Provider in case of such changes.
```
[the remark this answers is not in the document]
Comment (paulokow) [reply Domain Connect Spec Draft.adoc:1498]: I think this is more of an operational aspect. Service providers set it and same time are consuming this value, so I don't think there is a value of such recommendation.
For public templates one can also peek the version from the repo.

**Pull request merged**